code-challenges.test.js

```diff
@@ -13,39 +13,171 @@
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
 // a) Create a test with an expect statement using the variable provided.
+describe("uppercaseWords", () => {
+  it("takes in an array objects and returns an array with a sentence about each person with their name capitalized", () => {
+    const people = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+  expect(uppercaseWords(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+// combObj
+//   ✕ takes in an array objects and returns an array with a sentence about each person with their name capitalized
+
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+
+// b) Create the function that makes the test pass.
+
+//----------------- PSEUDO CODE -------------------
+// create an function with the name uppercaseWords give it a parameter of (arrOfPeople)
+// rest of the pseudo code is within the function so it would make sense. 
 
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+]  
+
+const uppercaseWords = (arrOfPeople) => {
+  //3 objects, enter each object
+  //2 properties, access [0].name, now have string
+  //capitalize and concatenation with first letter and rest of letter
+  //concatenation with name property, (after manip) and occ. property
+  //now completed string, push into arr
+  //return arr outside of the for loop
+  //for loop completes the work and pushes results into the array
+  const arr = []
+  for(let i = 0; i < arrOfPeople.length; i++){
+    //for each object, extract the name and occupation
+  let name = arrOfPeople[i].name
+  let occupation = arrOfPeople[i].occupation
+    //split name into two words
+  name = name.split(' ')
+  let firstName = name[0]
+  let toCapital = firstName[0].toUpperCase()
+  firstName = toCapital + firstName.substring(1)
+  let lastName = name[1]
+  toCapital = lastName[0].toUpperCase()
+  lastName = toCapital + lastName.substring(1)
+    //string = ${ add all words together}
+    //new arr []
+    //arr.push(string)
+  let str = `${firstName} ${lastName} is ${occupation}.`
+  arr.push(str)
+  
+    
+  }
+  
+  return arr
+}  
```

Comment:
Great pseudocode and logic looks excellent!  For a stretch, can you think of how to write this using higher-order functions instead of a for loop?